ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -570,10 +564,16 @@ public int compare(ReplicaId a, ReplicaId b) {
    * Create an {@link InstanceConfig} for the given node from the static cluster information.
    * @param node the {@link DataNodeId}
    * @param partitionToInstances the map of partitions to instances that will be populated for this instance.
+   * @param instanceToDiskReplicasMap the map of instances to the map of disk to set of replicas.
+   * @param referenceInstanceConfig the InstanceConfig used to set the fields that are not derived from the json files.
+   *                                These are the SEALED state and STOPPED_REPLICAS configurations. If this field is null,
+   *                                then these fields are derived from the json files. This can happen if this is a newly
+   *                                added node.
    * @return the constructed {@link InstanceConfig}
    */
-  private InstanceConfig createInstanceConfigFromStaticInfo(DataNodeId node,
-      Map<String, Set<String>> partitionToInstances) {
+  static InstanceConfig createInstanceConfigFromStaticInfo(DataNodeId node,
+      Map<String, Set<String>> partitionToInstances,
+      Map<String, Map<DiskId, SortedSet<Replica>>> instanceToDiskReplicasMap, InstanceConfig referenceInstanceConfig) {
```

Comment:
Why pass `instanceToDiskReplicasMap` as a parameter into this method? It seems `instanceToDiskReplicasMap` is global variable and doesn't get changed in this method.

Reply:
Wanted to make this method static so that it can be easily tested.